### DISCUSSION TITLE
Will ImHex use C++23 features that are only going to be available in GCC 13?

### DISCUSSION THREAD

**Discussion by Slackadays** [WerWolv/ImHex#884] [Get Help] [answered]
GCC 13 is bringing a few new C++20 features like std::format and some really useful functions from C++23, so I'm wondering if ImHex is going to switch to these as soon as they're available. However, this will also mean incompat with GCC 12 and below. [Clipboard](https://github.com/Slackadays/clipboard) is in a very similar situation, so I'm wondering what other projects are doing regarding this.

**Comment by WerWolv** [comment, marked as answer]
Absolutely! As soon as GCC 13 is available in some way on all the distros and systems we're targeting, I will start to switch everything over.
The vast majority of people won't need to compile imHex themselves anyways since we build nightly releases with every commit

**Comment by Slackadays** [comment]
I noticed that in the Minimal Requirements section, there's Ubuntu 22.04 which lags behind the new compiler releases. Does this mean the GCC 13 switch is going to take some time so that Ubuntu is compatible?

**Reply by WerWolv** [reply]
As of right now, the regular gcc release that ships with Ubuntu is like gcc 9 I think? But there's a second package, `gcc-12`, which was released not long after gcc 12 came out. So I assume the same will happen again when gcc 13 gets released
